homework/index.js

```diff
@@ -33,10 +33,81 @@
+
+        // CREATE EVENT LISTENER
+        select.addEventListener(
+          'change',
+          () => {
+            const selected = select.selectedIndex;
+            const repositoryValue = data[selected].name;
+            const repoLink = data[selected].html_url;
+            if (repositoryValue === null) {
+              document.getElementById('property_0').innerText = '-';
+            } else {
+              document.getElementById('property_0').innerText = repositoryValue;
+              document.getElementById('property_0').setAttribute('href', repoLink);
+              document.getElementById('property_0').setAttribute('target', '_blank');
+            }
+            const descriptionValue = data[selected].description;
+            if (descriptionValue === null) {
+              document.getElementById('property_1').innerText = '-';
+            } else {
+              document.getElementById('property_1').innerText = descriptionValue;
+            }
+            const forksValue = data[selected].forks;
+            if (forksValue === null) {
+              document.getElementById('property_2').innerText = '-';
+            } else {
+              document.getElementById('property_2').innerText = forksValue;
+            }
+            const updatedValue = new Date(data[selected].updated_at);
+            if (updatedValue === null) {
+              document.getElementById('property_3').innerText = '-';
+            } else {
+              document.getElementById('property_3').innerText = updatedValue;
+            }
+          },
+          false,
+        );
+
+        // SORT SELECTION MENU ALPHABETICALLY
+        const dataSorted = data.sort((a, b) => a.name.localeCompare(b.name));
+
+        // CREATE OPTIONS OF SELECTION MENU
+        dataSorted.forEach((repo, index) => {
+          createAndAppend('option', select, { text: repo.name, value: index });
+        });
+
+        const leftColumn = createAndAppend('div', root, { id: 'info_panel' });
+        const leftColumnTable = createAndAppend('table', leftColumn);
+        const leftColumnTableProperties = ['Repository :', 'Description :', 'Forks :', 'Updated :'];
+        const leftColumnTableFirstTr = createAndAppend('tr', leftColumnTable);
+        createAndAppend('td', leftColumnTableFirstTr, { text: leftColumnTableProperties[0] });
+        const linkRepo = createAndAppend('td', leftColumnTableFirstTr);
+        createAndAppend('a', linkRepo, { id: `property_${0}` });
+
+        // CREATE TABLE ELEMENTS OF LEFT COLUMN
+        for (let j = 1; j < 4; j++) {
+          const leftColumnTableTr = createAndAppend('tr', leftColumnTable);
+          createAndAppend('td', leftColumnTableTr, { text: leftColumnTableProperties[j] });
+          createAndAppend('td', leftColumnTableTr, { id: `property_${j}` });
+        }
+
+        // GET DEFAULT VALUES OF LEFT COLUMN AT PAGE OPENING
+        document.getElementById('property_0').innerText = data[0].name;
```

Comment:
Small suggestion: perhaps write a function that can assign this value, so you don't have to copy setting text, href and target. E.g. `setLinkAttributes(data[0].name, data[0].html_url, '_blank');`

Reply:
Written a function for this section.